curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo/64fb29348a60361ccd45c1e2.md

```diff
@@ -0,0 +1,404 @@
+---
+id: 64fb29348a60361ccd45c1e2
+title: Step 46
+challengeType: 0
+dashedName: step-46
+---
+
+# --description--
+
+`localStorage` has methods you can use for saving, retrieving, and deleting items. The items you're saving could be any JavaScript data type.
+
+For example, the `setItem()` method saves an item and the `getItem()` method retrieves the item. If you want to delete an item, you can use the `removeItem()` method, or `clear()` to delete all items in the storage.
+
+To save an item, you use `localStorage.setItem()`, pass in the key as a string, and the data you want to save.
```

Comment:
We should show a code example here to show the syntax  instead of explaining the syntax to campers 
It will be easier for beginners. We could use something like this below: 


Here is an example of how to set an item in `localStorage`

````
```
 localStorage.setItem("bgcolor", "red");
```
````